### ISSUE TITLE
3.0 - Improve tax rates add row styling

### ISSUE THREAD

**Issue by davidsherlock** [easydigitaldownloads/easy-digital-downloads#8405]
### Expected behavior
When I add a tax rate that is not country-wide, I expect the form styling to be pleasing from a UX perspective.

### Actual behavior

Form looks like this:

<img width="1036" alt="Screenshot 2021-02-17 at 10 41 15" src="https://user-images.githubusercontent.com/22668877/108171703-f5a96e00-710c-11eb-871a-5bcaaacb36b6.png">

In my opinion, a few things need fixed here:

1. The label spacing between the checkbox and label is too small.
2. The height of the input fields/submit button is incongruent.
3. The location of the "state" select/text input when loaded via AJAX is messy.

### Steps to reproduce the behavior
1) Disable the "Apply to whole country" select box.
2) You will see the state box appears directly below it.

### Information (if a specific version is affected):
PHP Version: 8.0

EDD Version (or branch): 3.0-beta

WordPress Version: 5.7

**Comment by spencerfinnell** [comment]
I agree -- not sure why I felt it needed to be stuffed in to the table footer originally. A simple stacked form below would work, or a modal form similar to what is used when adding a Download to an Order.

**Comment by davidsherlock** [comment]
@spencerfinnell I second the modal approach, definitely would make for a much cleaner UX.

**Comment by spencerfinnell** [comment]
Does anyone have thoughts on where the "Add Rate" button to launch the modal would go?

**Comment by davidsherlock** [comment]
<img width="1057" alt="Screenshot 2021-02-26 at 19 17 37" src="https://user-images.githubusercontent.com/22668877/109325722-518b9980-7867-11eb-9afc-cd31301d5ef1.png">

This feels organic to me, thoughts?

**Comment by spencerfinnell** [comment]
With a bit of other UI cleanup:

<img width="877" alt="Screen Shot 2021-02-26 at 11 21 38 AM" src="https://user-images.githubusercontent.com/491124/109326198-d86b5280-7824-11eb-81cc-079fd59d3fd1.png">

**Comment by davidsherlock** [comment]
@ashleyfae I tested this and it works good with the model approach.

**Comment by ashleyfae** [comment]
I've set this to 3.0.5 just because it's got quite a lot in it and will need more testing and merge conflict  fixing.
